File: integrations/src/main/java/framework/ThirdPartyHandler.java
package framework;

import framework.config.ThirdPartyServiceList;
import framework.core.data.RunConfig;

public abstract class ThirdPartyHandler {
    protected RunConfig runConfig;

    public abstract ThirdPartyServiceList getKey();

    public void setupConfig(RunConfig runConfig) {
        this.runConfig = runConfig;
    }
}

File: integrations/src/main/kotlin/framework/ThirdPartyHandler.kt
package framework

import framework.config.ThirdPartyServiceList
import framework.core.data.RunConfig

abstract class ThirdPartyHandler {
    lateinit var runConfig: RunConfig
    abstract fun getKey(): ThirdPartyServiceList
    fun setupConfig(runConfig: RunConfig) {
        this.runConfig = runConfig
    }
}
